### ISSUE TITLE
wish: throttle "restartall"

### ISSUE THREAD

**Issue by markstos** [foreversd/forever#603]
I have several produces managed by `forever` which work as a pool. The reverse proxy in front of them can gracefully handle some members of the pool being offline. 

However, it appears that when 'forever restartall' is used, all my node instances are going offline at once, and the app temporarily becomes unavailable.  This behavior is nice in some ways in that it's fast and keeps all the nodes together in a consistent state. 

Still, it would be nice to have an option for a throttled version of restartall that restarts one at a time, perhaps with a pause. 

As a workaround, I could write a script to query `forever list` and then call `forever restart` on the node instances one by one.

**Comment by markstos** [comment]
I'm not sure what happened to the commit above. Here's an alternate pull request which implements just the rolling-restart request:

https://github.com/foreverjs/forever.git

**Comment by markstos** [comment]
Could I have some updated feedback on the PR I proposed above. It proposes always using a rolling restart for the "restart" cases with multiple nodes, to avoid downtime when all the nodes have stopped, but none have started yet. 

This mirrors the [zero downtime upgrade](https://www.nginx.com/faqs/how-does-zero-downtime-configuration-testingreload-in-nginx-plus-work/) approach that Nginx uses. I'll note that Nginx does not offer configuration options for how a restart happens. The rolling zero-downtime approach is the only option. I recommend emulating Nginx's successful design here. 

If you really want to take the entire cluster offline before bring up workers with a new configuration, there is already an option to do this, just like there is in Nginx: Just issue a `stop` command followed by a `start` command.

**Comment by markstos** [comment]
I'll note that the StrongLoop `slc` tool uses rolling starts. Search for "rolling restart" on [this page](http://strong-pm.io/prod/) to see it mentioned. 

Also note this chart comparing `pm2` with ` StrongLoop Process Manager` and `forever`.  There's a line item for "Rolling Restarts" as a desirable feature which the two competing options have, but `forever` does not: http://strong-pm.io/compare/

`pm2` offers [several options](http://pm2.keymetrics.io/docs/usage/cluster-mode/#reload-without-downtime):
 
  * stop/start
  * restart (sounds like it's implemented like stop/start
  * reload (sounds like a rolling restart)
  * gracefulReload (sounds like waiting for connections to die before starting) 

Implementing the rolling-start as "reload" instead of "restart" as option, but I think it could just be considered an improvement to "restart", without requiring people to figure which of 4 different options for restarting their app is best.
